pkg/cloudprovider/providers/vsphere/vsphere_util.go

```diff
@@ -63,3 +65,8 @@ func GetgovmomiClient(cfg *VSphereConfig) (*govmomi.Client, error) {
 	client, err := newClient(context.TODO(), cfg)
 	return client, err
 }
+
+func IsIpv4Address(str string) bool {
+	ip := net.ParseIP(str)
```

Comment:
can you use To4() method from net package.
`ip := net.ParseIP(str).To4()`

For ipv6 address `ip` will be set to nil.